eva_submission/submission_qc_checks.py

```diff
@@ -0,0 +1,316 @@
+        failed_files = {}
+        for file in vcf_files:
+            accessioning_log_files = glob.glob(f"{self.path_to_data_dir}/00_logs/accessioning.*{file}*.log")
+            if accessioning_log_files:
+                # check if any variants were skippped while accessioning
+                variants_skipped = self.check_if_variants_were_skipped(accessioning_log_files[0])
+                if variants_skipped:
+                    failed_files[file] = f"{variants_skipped} variants skipped"
+            else:
+                failed_files[file] = f"error : No accessioning file found for {file}"
+
+        self._variants_skipped_accessioning_check_result = "PASS" if not failed_files else "FAIL"
+        report = f"""
+                pass: {self._variants_skipped_accessioning_check_result}"""
+        if failed_files:
+            report += f"""
+                failed_files:"""
+            for file, value in failed_files.items():
+                report += f"""
+                    {file} - {value}"""
+
+        return report
+
+    def check_if_browsable_files_entered_correctly_in_db(self, vcf_files):
+        browsable_files_from_db = self.get_browsable_files_for_study()
+        missing_files = set(vcf_files) - set(browsable_files_from_db)
+        self._browsable_files_check_result = "PASS" if len(missing_files) == 0 else "FAIL"
+
+        return f"""
+            pass : {self._browsable_files_check_result}
+            expected files: {vcf_files}
+            missing files: {missing_files if missing_files else 'None'}"""
+
+    def run_qc_checks_for_submission(self):
+        """Collect information from different qc methods and write the report."""
+        vcf_files = []
+        for analysis_data in self.analyses.values():
+            for v_files in analysis_data['vcf_files'].values():
+                vcf_files.append(os.path.basename(v_files['output_vcf_file']))
+
+        browsable_files_report = self.check_if_browsable_files_entered_correctly_in_db(vcf_files)
+
+        # No accessioning check is required for human
+        if self.taxonomy != 9606:
+            accessioning_job_report = self.check_if_accessioning_completed_successfully(vcf_files)
+            variants_skipped_report = self.check_if_variants_were_skipped_while_accessioning(vcf_files)
+
+        variant_load_report = self.check_if_variant_load_completed_successfully(vcf_files)
+
+        ftp_report = self.check_all_browsable_files_are_available_in_ftp(vcf_files)
+
+        study_dev_report = self.check_if_study_appears_in_dev()
+
+        study_dev_metadata_report = self.check_if_study_appears_in_dev_metadata()
+
+        report = f"""
+        QC Result Summary:
+        ------------------
+        Browsable files check: {self._browsable_files_check_result}
+        Accessioning job check: {self._accessioning_job_check_result}
+        Variants Skipped accessioning check: {self._variants_skipped_accessioning_check_result}
```

Comment:
These two variables (and the corresponding report variables) will be undefined for human studies and will raise errors... either don't include them in the report in that case, or set them to reasonable default values for human studies.